test/unit/query_context_cache_test.js

```diff
@@ -0,0 +1,221 @@
+      this.qcc.merge(this.expectedIDs[i], this.expectedReadTimestamp[i], this.expectedPriority[i], Context);
+    }
+    this.qcc.syncPriorityMap();
+  };
+    
+  this.initCacheWithDataInRandomOrder = function () {
+    this.qcc = new QueryContextCache(this.MAX_CAPACITY);
+    this.expectedIDs = [];
+    this.expectedReadTimestamp = [];
+    this.expectedPriority = [];
+    for (let i = 0; i < this.MAX_CAPACITY; i++) {
+      this.expectedIDs[i] = this.BASE_ID + i;
+      this.expectedReadTimestamp[i] = this.BASE_READ_TIMESTAMP + i;
+      this.expectedPriority[i] = this.BASE_PRIORITY + i;
+    }
+  
+    this.qcc.merge(this.expectedIDs[3], this.expectedReadTimestamp[3], this.expectedPriority[3], this.CONTEXT);
+    this.qcc.merge(this.expectedIDs[2], this.expectedReadTimestamp[2], this.expectedPriority[2], this.CONTEXT);
+    this.qcc.merge(this.expectedIDs[4], this.expectedReadTimestamp[4], this.expectedPriority[4], this.CONTEXT);
+    this.qcc.merge(this.expectedIDs[0], this.expectedReadTimestamp[0], this.expectedPriority[0], this.CONTEXT);
+    this.qcc.merge(this.expectedIDs[1], this.expectedReadTimestamp[1], this.expectedPriority[1], this.CONTEXT);
+    this.qcc.syncPriorityMap();
+  };
+
+  this.assertCacheData = function () {
+    this.assertCacheDataWithContext(this.CONTEXT);
+  };
+  
+  this.assertCacheDataWithContext = function (Context) {
+    const size = this.qcc.getSize();
+    assert.strictEqual(size,this.MAX_CAPACITY);
+    const elements = Array.from(this.qcc.getElements());
+    for (let i = 0; i < size; i++) {
+      assert.strictEqual(this.expectedIDs[i], elements[i].getId());
+      assert.strictEqual(this.expectedReadTimestamp[i], elements[i].getReadTimestamp());
+      assert.strictEqual(this.expectedPriority[i], elements[i].getPriority());
+      assert.strictEqual(Context, elements[i].getContext());
+    }
+  };
+}
+
+describe('QueryContextCacheTest', function () {
+  const testingQcc = new TestingQCC();
+
+  /** Test for empty cache */
+  it('testIsEmpty',function () {
+    testingQcc.initCache();
+    assert.strictEqual(testingQcc.qcc.getSize(), 0);
+  });
+  
+  it('testWithSomeData',function () {
+    testingQcc.initCacheWithData();
+
+    // Compare elements
+    testingQcc.assertCacheData();
+  });
+  
+  it('testWithSomeDataInRandomOrder',function () {
+    testingQcc.initCacheWithDataInRandomOrder();
+
+    // Compare elements
+    testingQcc.assertCacheData();
+  });
+  
+  it('testMoreThanCapacity',function () {
+    testingQcc.initCacheWithData();
+  
+    // Add one more element at the end
+    const i = testingQcc.MAX_CAPACITY;
+    testingQcc.qcc.merge(testingQcc.BASE_ID + i, testingQcc.BASE_READ_TIMESTAMP + i, testingQcc.BASE_PRIORITY + i, testingQcc.CONTEXT);
+    testingQcc.qcc.syncPriorityMap();
+    testingQcc.qcc.checkCacheCapacity();
+  
+    // Compare elements
+    testingQcc.assertCacheData();
+  });
+  
+  it('testUpdateTimestamp',function () {
```

Comment:
cannot we make those tests named and written in more explicit ways? we are just comparing and merge list of simple java script objects

maybe even let's add some parameterized tests to check what's the initial state of the cache, what is the incoming query context and what is the expected query context in the cache

Reply:
I fixed the names of the tests. If you do not mind, could you please elaborate on "more explicit ways" for testing? Do you mean that we need to create detailed tests? If you have a reference, please let me know.

Reply:
I mean that when we have set of test cases we can name them all to show for what is each of the tests

Reply:
Oh, I see. Then, I think this is done.